src/sortTemplate.ts

```diff
@@ -0,0 +1,297 @@
+}
+
+async function sortGenericDeep<T, TChild>(list: T[], childSelector: (value: T) => TChild[] | undefined, sortSelector: (value: TChild) => string, spanSelector: (value: TChild) => language.Span, template: DeploymentTemplate): Promise<void> {
+    for (let item of list) {
+        let children = childSelector(item);
+        if (children !== undefined) {
+            await sortGeneric(children, sortSelector, spanSelector, template);
+        }
+    }
+}
+
+function getNameFromResource(value: Json.Value): string {
+    let objectValue = Json.asObjectValue(value);
+    if (objectValue === undefined) {
+        return "";
+    }
+    let nameProperty = objectValue.getPropertyValue("name");
+    if (nameProperty === undefined) {
+        return "";
+    }
+    let name = nameProperty.toFriendlyString();
+    return name;
+}
+
+function getResourcesFromResource(value: Json.Value): Json.Value[] | undefined {
+    let objectValue = Json.asObjectValue(value);
+    if (objectValue === undefined) {
+        return undefined;
+    }
+    let resources = Json.asArrayValue(objectValue.getPropertyValue("resources"));
+    if (resources === undefined) {
+        return undefined;
+    }
+    return resources.elements;
+}
+
+function joinTexts(texts: String[], intendentTexts: String[]): string {
+    let output: string = "";
+    for (let index = 0; index < texts.length - 1; index++) {
+        output = `${output}${texts[index]}${intendentTexts[index]}`;
+    }
+    output = `${output}${texts[texts.length - 1]}`;
+    return output;
+}
+
+function getIndentTexts<T>(list: T[], spanSelector: (value: T) => language.Span, document: vscode.TextDocument): String[] {
+    let indentTexts: String[] = [];
+    for (let index = 0; index < list.length - 1; index++) {
+        let span1 = spanSelector(list[index]);
+        let span2 = spanSelector(list[index + 1]);
+        let intendentText = document.getText(new vscode.Range(document.positionAt(span1.afterEndIndex), document.positionAt(span2.startIndex)));
+        indentTexts.push(intendentText);
+    }
+    return indentTexts;
+}
+
+function getRange(span: language.Span, document: vscode.TextDocument): vscode.Range {
+    return new vscode.Range(document.positionAt(span.startIndex), document.positionAt(span.afterEndIndex));
+}
+
+function getText(span: language.Span, document: vscode.TextDocument): String {
+    let range = getRange(span, document);
+    let text = document.getText(range);
+    return text;
+}
+
+function getSelection(start: language.Span, end: language.Span, document: vscode.TextDocument): vscode.Selection {
+    let startIndex = start.startIndex;
+    let endIndex = end.afterEndIndex;
+    let selection = new vscode.Selection(document.positionAt(startIndex), document.positionAt(endIndex));
+    return selection;
+}
+
+function arraysEqual<T>(a: T[], b: T[]): boolean {
+    if (a === b) {
+        return true;
+    }
+    if (a == null || b == null) {
```

Comment:
```suggestion
    if (!a || !b) {
```

Reply:
If I make the change I get the following warning from tslint

This type is not allowed in the operand for the '!' operator because it is always truthy. Allowed types are boolean, null-union, undefined-union, or string. (strict-boolean-expressions)

Reply:
Fixed

Reply:
Fixed